src/main/java/seedu/address/model/student/Phone.java

```diff
@@ -21,8 +21,10 @@
      * @param phone A valid phone number.
      */
     public Phone(String phone) {
-        requireNonNull(phone);
-        checkArgument(isValidPhone(phone), MESSAGE_CONSTRAINTS);
+        if (!phone.equals("")) {
```

Comment:
Actually this check is not needed since the regex does this for you (see isValidPhone())

Reply:
Note for @PotatoCombat : double check if the regex does this already.